projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -669,7 +669,15 @@ def __verify_payload_format(payload_dict: dict):
         # Check if payload dict is valid json
         # TODO: optimize the check - we should not need to serialize the payload every time
         try:
-            json.dumps(payload_dict)
+            from decimal import Decimal
```

Comment:
Tests? Any change must have an automated test written or a good reason not to have one.

Reply:
The __verify_payload_format changes are covered by the existing ingestion tests (I verified this in the coverage), added a unittest for the DecimalJsonEncoder class.

Reply:
> The __verify_payload_format changes are covered by the existing ingestion tests

The fact that we have a bugfix in progress shows that the existing tests do *not* cover the error. Otherwise we'd not be fixing it :) 

The test coverage is misleading. It simply shows that we have visited that line of code. But we have visited with another argument (clearly not decimal). We need tests that verifies the behaviour of the ingest when it's passed decimal values.

Reply:
This is correct, I've amended of the tests to include a Decimal value.